### PR TITLE
Enable Lint Rules: bare-return & empty-lines

### PR DESCRIPTION
**Pull request by FlamingSaint**
## Which problem is this PR solving?
-  Partial Fix for #5506

## Description of the changes
- Enabled bare-return in revive linter. 
- Removed naked returns.
- Enabled empty-lines in revive linter.

## How was this change tested?
- `make lint` `make test`

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: pkg/httpmetrics/metrics.go [hunk 1]
```diff
@@ -149,5 +149,5 @@ func (r *requestDurations) buildTimer(metricsFactory metrics.Factory, key record
 			"method": key.method,
 		},
 	})
-	return
+	return out
```
Comment (yurishkuro) [inline pkg/httpmetrics/metrics.go:152]: We can remove the name "out" from the signature

File: pkg/kafka/auth/plaintext.go [hunk 2]
```diff
@@ -44,7 +44,10 @@ func (x *scramClient) Begin(userName, password, authzID string) (err error) {
 // called repeatedly until it errors or `Done` returns true.
 func (x *scramClient) Step(challenge string) (response string, err error) {
```
Comment (yurishkuro) [inline pkg/kafka/auth/plaintext.go:45]: Remove output names

File: pkg/kafka/auth/plaintext.go [hunk 3]
```diff
@@ -44,7 +44,10 @@ func (x *scramClient) Begin(userName, password, authzID string) (err error) {
...
 	response, err = x.ClientConversation.Step(challenge)
```
Comment (yurishkuro) [inline pkg/kafka/auth/plaintext.go:46]: Just do a return directly

File: pkg/gzipfs/gzip.go [hunk 4]
```diff
@@ -60,7 +60,7 @@ func (f *file) Read(buf []byte) (n int, err error) {
 		return n, io.EOF
 	}
 	f.offset += n
-	return
+	return n, nil
```
Comment (yurishkuro) [inline pkg/gzipfs/gzip.go:63]: Remove out names from signature

File: pkg/httpmetrics/metrics.go (hunk 1 above)
Comment (FlamingSaint) [reply pkg/httpmetrics/metrics.go:152]: The defer function on top requires the named output as its assigning a value to it

**Review by yurishkuro — APPROVED** [review]

**Review by yurishkuro — APPROVED** [review]

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**